signing.go

```diff
@@ -156,6 +168,13 @@ func (ctx *genericSigner) Sign(payload []byte) (*JsonWebSignature, error) {
 	return obj, nil
 }
 
+// AddNonces provides or updates a nonce pool to the first recipients.
+// After this method is called, the signer will consume one nonce per
+// signature, returning an error if there are no nonces left in the pool.
+func (ctx *genericSigner) AddNonces(nonces []string) {
```

Comment:
Rather than having a nonce pool that gets exhausted, would it be preferable to simply pass the signer a callback function for retrieving nonces? E.g. something like `SetNonceProvider(func getNonce() string)` and have the signer call `getNonce` when one is required.